recipy/log.py

```diff
@@ -293,17 +297,44 @@ def output_file_diffs():
     if not option_set('data', 'file_diff_outputs'):
         return
 
-    db = open_or_create_db()
-    diffs_table = db.table('filediffs')
-    diffs = diffs_table.search(Query().run_id == RUN_ID)
+    encodings = ['utf-8', 'latin-1']
+
+    with open_or_create_db() as db:
+        diffs_table = db.table('filediffs')
+        diffs = diffs_table.search(Query().run_id == RUN_ID)
+
     for item in diffs:
-        diff = difflib.unified_diff(open(item['tempfilename']).readlines(),
-                                    open(item['filename']).readlines(),
-                                    fromfile='before this run',
-                                    tofile='after this run')
-        diffs_table.update({'diff': ''.join([l for l in diff])},
-                           eids=[item.eid])
+        if option_set('general', 'debug'):
+            print('Storing file diff for "%s"' % item['filename'])
+
+        lines1 = None
+        lines2 = None
+        for enc in encodings:
+            try:
+                with codecs.open(item['tempfilename'], encoding=enc) as f:
+                    lines1 = f.readlines()
+            except UnicodeDecodeError:
+                pass
```

Comment:
Personally, `pass`ing multiple exceptions is a code smell for me, especially since the consequences of a thrown exceptions are detected further down below with the `if lines...` statement.
How about this:
Add an `open_failed = False` variable before trying to open the files. In both except branches here and in L323, replace `pass` with `open_failed = True`. Then, replace `if lines1 is not None and lines2 is not None:` with `if open_failed:`.
This way, you don't have ignored the exceptions, you have coupled the cause (opening failed) with the effect (show warning message) more strongly via `open_failed`, and the name of the sentinel makes it clear what you intent to test in the if statement. :-)